sabnzbd/getipaddress.py

```diff
@@ -110,16 +110,15 @@ def publicipv4():
     public_ipv4 = None
     try:
         ipv4_found = False
-        # we only want IPv4 resolving, so socket.AF_INET:
-        result = addresslookup4(sabnzbd.cfg.selftest_host())
-    except (ValueError, socket.error, multiprocessing.context.TimeoutError):
-        # something very bad: no urllib2, no resolving of selftest_host, no network at all
-        # Or strange DSM problem: https://github.com/sabnzbd/sabnzbd/issues/2008
+        # look up IPv4 addresses of selftest_host
+        lookup_result = addresslookup4(sabnzbd.cfg.selftest_host())
+    except:
```

Comment:
Because the user reasonably expects an application to shut down upon hitting ctrl-c. Catching BaseException will prevent that from happening or at least require multiple attempts to achieve the desired effect (making the application appear unresponsive in the meantime), and instead in this case only abort a single function call that just happens to be waiting for a timeout.

Reply:
Thanks!

Reply:
Thanks!